File: 알고리즘/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        a = list(s)
        b = list(t)
        a.sort()
        b.sort()
        
        if(len(a)!=len(b)):
            return False
        
        for i in range(len(a)):
            if(a[i]!=b[i]):
                return False
            
        return True

    def isAnagram2(self, s: str, t: str) -> bool:
        counter =[0 for _ in range(26)]

        for i in range(len(s)):
            counter[s[i]-'a']+=1
            counter[t[i]-'a']-=1

        for ele in counter:
            if(ele!=0):
                return False
        return True  
    

File: 알고리즘/Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());
        
        if(s.size()!=t.size()) return false;
        
        for(int i=0; i<s.size(); i++){
            if(s[i]!=t[i]) return false;
        }
        
        return true;
        
    }

    bool isAnagram2(string s, string t){
        if(s.size()!=t.size()) return false;

        int counter[26] = {0,};

        for(int i=0; i<s.size(); i++){
            counter[s[i]-'a']+=1;
            counter[t[i]='a']-=1;
        }

        for(auto ele : counter){
            if(ele!=0) return false;
        }
        
        return true;
    }
};